### DISCUSSION TITLE
Task 14: iterative Weight Windows via Python API  - 4_sphere_iterative_per_batch_ww

### DISCUSSION THREAD

**Discussion by alanottrenfusion** [fusion-energy/neutronics-workshop#299] [Q&A]
Hey guys !  
I just encountered a problem with the "Task 14: iterative Weight Windows via Python API  - 4_sphere_iterative_per_batch_ww" update_magic(ww_tally) process.
It seems using the openmc 014, the whole task is running perfectly. (the 2 calculations are about 1 minute)
I updated to the openmc 015, the calculation update_magic  blocked in the second batch with no specific message.
Alan

**Comment by shimwell** [comment]
Thanks for bringing this up.
We might have to look at openmc release notes and api changes to solve this one.
I shall see if I can reproduce this

**Comment by shimwell** [comment]
Oh just to mention the workshop is tested on the develop branch of openmc. Might have more luck using that instead of version 0.15.

**Reply by alanottrenfusion** [reply]
Hi ! Thanks for your reply.
Do you mean here: https://github.com/openmc-dev/openmc/tree/develop/tests
I can't find it. Maybe in another placeholder ?

**Reply by shimwell** [reply]
I mean we run CI on this neutronics-workshop repo and in that CI we use the openmc current develop branch which is currently 59 commits ahead of version 0.15. So it might be worth trying the develop branch of openmc which is the version in the neutronics-workshop docker file

**Reply by alanottrenfusion** [reply]
Thanks ! i will try it out ! : )
I got back to the openmc 0.14 to use the magic feature and perform efficient calculation. It works for now.
Is it possible to install the dev version with conda/mamba or with spack ?
Thanks a lot for your help !

**Reply by shimwell** [reply]
conda / mamba only has released versions distributed, so no develop version there https://anaconda.org/conda-forge/openmc/files

I am not sure about spack but here is a link to the install docs for openmc which might contain more info
https://docs.openmc.org/en/stable/quickinstall.html#manually-installing-from-source

**Comment by alanottrenfusion** [comment]
Hi Mr Shimwell,
i finaly got openmc v0.15.1 installed from source:
----------------------------------------------------
openmc --version
OpenMC version 0.15.1
Git SHA1: 7552123c762a13be49269bf3c5bf393635109766
Copyright (c) 2011-2024 MIT, UChicago Argonne LLC, and contributors
MIT/X license at <https://docs.openmc.org/en/latest/license.html>
Build type:            RelWithDebInfo
Compiler ID:           GNU 11.4.0
MPI enabled:           no
Parallel HDF5 enabled: no
PNG support:           yes
DAGMC support:         no
libMesh support:       no
MCPL support:          no
NCrystal support:      no
Coverage testing:      no
Profiling flags:       no
UWUW support:          no
----------------------------------------------------

but it leads to the same results => when excecuting the loop with update_magic, the run get stuck in batch number 2:

the output 
----------------------------------------------------

The OpenMC Monte Carlo Code
Copyright | 2011-2024 MIT, UChicago Argonne LLC, and contributors
License | https://docs.openmc.org/en/latest/license.html
Version | 0.15.1-dev
Git SHA1 | 7552123c762a13be49269bf3c5bf393635109766
Date/Time | 2024-11-05 11:00:52
OpenMP Threads | 15

 Reading settings XML file...
 Reading cross sections XML file...
 Reading materials XML file...
 Reading geometry XML file...
 Reading H1 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/H1.h5
 Reading H2 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/H2.h5
 Reading C12 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/C12.h5
 Reading C13 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/C13.h5
 Reading O16 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/O16.h5
 Reading O17 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/O17.h5
 Reading O18 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/O18.h5
 Reading Na23 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Na23.h5
 Reading Mg24 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Mg24.h5
 Reading Mg25 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Mg25.h5
 Reading Mg26 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Mg26.h5
 Reading Al27 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Al27.h5
 Reading Si28 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Si28.h5
 Reading Si29 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Si29.h5
 Reading Si30 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Si30.h5
 Reading K39 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/K39.h5
 Reading K40 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/K40.h5
 Reading K41 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/K41.h5
 Reading Ca40 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Ca40.h5
 Reading Ca42 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Ca42.h5
 Reading Ca43 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Ca43.h5
 Reading Ca44 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Ca44.h5
 Reading Ca46 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Ca46.h5
 Reading Ca48 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Ca48.h5
 Reading Fe54 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Fe54.h5
 Reading Fe56 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Fe56.h5
 Reading Fe57 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Fe57.h5
 Reading Fe58 from
 /mnt/c/Users/aot/workspace/data/endfb80/endfb-viii.0-hdf5/neutron/Fe58.h5
 Minimum neutron data temperature: 0 K
 Maximum neutron data temperature: 1.7976931348623157e+308 K
 Reading tallies XML file...
 Preparing distributed cell instances...
 Reading plot XML file...
 Writing summary.h5 file...
 Maximum neutron transport energy: 20000000 eV for H1

 ===============>     FIXED SOURCE TRANSPORT SIMULATION     <===============

 Simulating batch 1
 Simulating batch 2
----------------------------------------------------

Do you have any idea where this beviour can come from ?

**Comment by jon-proximafusion** [comment]
I've just run this task within the docker container and it appears to work fine for me.

Did you change anything in the task?

If not then I guess there is a difference between the environment in the docker container and your system.

You can take a peak at the [docker files](https://github.com/fusion-energy/neutronics-workshop/tree/main/.devcontainer) to see how I have installed things and perhaps there is a difference somewhere
